pkg/sql/catalog/descriptor.go

```diff
@@ -88,6 +88,8 @@ type DescriptorBuilder interface {
 	// upgrade migrations
 	RunRestoreChanges(descLookupFn func(id descpb.ID) Descriptor) error
 
+	SetRawBytesInDescriptor(rawBytes []byte)
```

Comment:
Can we pass the raw bytes to the `DescriptorBuilder` constructors instead? Either you create the builder from another descriptor and that byte slice is available there, or you create it from a protobuf and the bytes are usually not far. In any case, we have to pass them along.

I think this PR might benefit from being rebased on https://github.com/cockroachdb/cockroach/pull/88786 which tightens how these builders can be created.

Reply:
The difficulty would be when we upsert a descriptor to the `uncommitted` descriptor set, we will need to update this `rawBytesInStorage` field, instead of passing it along. This is the main reason I choose to have a setter on this field.